File: mcspring-build/mcspring-plugin-manager/src/main/java/in/kyle/mcspring/manager/commands/CommandOp.java
package in.kyle.mcspring.manager.commands;

import org.bukkit.command.CommandSender;
import org.springframework.stereotype.Component;

import in.kyle.mcspring.command.Command;
import in.kyle.mcspring.subcommands.PluginCommand;

@Component
class CommandOp {
    
    @Command(value = "op",
             description = "Toggle yourself or another players OP status",
             usage = "/op <player>?")
    void op(PluginCommand command) {
        command.withPlayer(s -> String.format("Player %s not found", s));
        command.then(this::toggleOp);
        
        command.otherwise(this::toggleOp);
    }
    
    String toggleOp(CommandSender target) {
        target.setOp(!target.isOp());
        return String.format("%s is %s op", target.getName(), target.isOp() ? "now" : "no longer");
    }
}

File: mcspring-build/mcspring-plugin-manager/src/main/kotlin/in/kyle/mcspring/manager/commands/CommandOp.kt
package `in`.kyle.mcspring.manager.commands

import `in`.kyle.mcspring.commands.dsl.commandExecutor
import `in`.kyle.mcspring.commands.dsl.mcspring.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.springframework.stereotype.Component

@Component
internal class CommandOp {

    @Command(
            value = "op",
            description = "Toggle yourself or another players OP status",
            usage = "/op <player>?"
    )
    fun op() = commandExecutor {
        val target = playerArg {
            default { sender as? Player }
            invalid { message("Player $it not found") }
        }
        then { message(toggleOp(target)) }
    }

    private fun toggleOp(target: CommandSender): String {
        target.isOp = !target.isOp
        val modifier = if (target.isOp) "now" else "no longer"
        return "${target.name} is $modifier op"
    }
}
